### DISCUSSION TITLE
Task Priority System

### DISCUSSION THREAD

**Discussion by latraing** [meilisearch/product#709] [Feedback & Feature Proposal]
Hello,

I would like to propose the idea of having some sort of task prioritization feature in order to have certain tasks get processed first. With the way my team has ingestion into our Meilisearch database set up, we find that there can be many situations when there are several tasks to put documents into some rather large indexes (in the millions) that have to be processed first before it reaches the task that we want done as soon as possible. If we could have a system of putting priority numbers on tasks themselves, Meilisearch would then be able to pick up the tasks we really want done first.
This can be very useful if we know ahead of time certain tasks to small indexes finish really quickly, so there can be some manual setting of priority on our side to let Meilisearch know to pick those up before going back to processing the larger tasks. If the task queue is backlogged, it would be really convenient to have a way to 'skip' the queue, especially for small updates to the database that end users might expect to be updated relatively quickly. I think having more control over the order Meilisearch processes tasks would greatly improve the overall experience.

Thank you.

**Comment by macraig** [comment]
Hi @latraing , apologies for the delayed reply. Thank you for opening a discussion on this topic. We're currently discussing ways of improving how the task queue works. We'll take your idea into account during our discussions. 

Once again thank you for your contribution!

**Comment by tacman** [comment]
Yeah, this would be great.  I'm indexing 6 million documents in one index, indexing in batches of 10,000, and in another I have just 100, but it appears the second one has to wait its turn.    I know some of the operations need to be run in order, but it'd be up to the programmer to figure that out.
